584 multiple repos software

dmijatovic opened the pull request:
# Support multiple repositories software

Closes #584

Changes proposed in this pull request:
* Add source code section to edit software pages. The section supports adding multiple repositories to one software entry. The platform drop-down is enabled only in the cases where we cannot detect platform. The user can only add or remove repository, editing repository is not possible because change of URL means that it is different repository.
* Adapt background services section. The background services reports only the status of scrapers of the first repository.   
* Add repositories section to rsd admin pages. Here rsd admin can disable scrapers for specific repository by providing the text/explanation. Rsd admin can remove repository from RSD which will remove it from all software entries too.
* Fix scraper requests to repository_url table (replaced software column with id in request)
* Chore: rename frontend config object to cfg due to naming collision with next app router naming conventions

How to test:
* `docker compose build && docker compose up` to build and start without test data.
* Add new software and create all different entries: 
    * Github: https://github.com/research-software-directory/RSD-as-a-service
    * Gitlab: https://gitlab.com/cyber5k/mistborn
    * Bitbucket: https://bitbucket.org/dmijatovic/160301/src/master/
    * 4TU: https://data.4tu.nl/v3/datasets/058eb2e5-060e-4176-9601-9a5af3ac60d5.git
    * Codeberg: https://codeberg.org/forgejo/k8s-cluster
* Add also few package managers
* Check background service messages.
* Confirm the scrapers are running properly and without errors. 
* Login as rsd admin and navigate to admin/repositories page. Search for repo and change code platform or add reason for disabling scraper. Confirm that reason for disabling is shown on the admin item but also on the software item.
* Delete repository as admin and confim that repo is deleted from all software items where it was used.
* Examine info messages on Links & metadata edit software page. RepoURL info is removed.
* Examine info message on Source code edit software page.

## Example source code section software page
<img width="1612" height="913" alt="image" src="https://github.com/user-attachments/assets/36c369ee-6a13-4d8c-bd23-84a95c6689fc" />

## Example source code section on edit software page
<img width="1560" height="909" alt="image" src="https://github.com/user-attachments/assets/3b8e66b2-930d-4f1a-b7e5-38f001943618" />

## Example repository admin section
<img width="1550" height="1145" alt="image" src="https://github.com/user-attachments/assets/97163546-cb62-4815-8702-1639a60f5220" />


PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [ ] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

ewan-escience committed:
wip: normalise repository_url and allow multiple entries per software

dmijatovic committed:
feat: enable defining multiple repositories for one software item
feat: add repository section to admin pages
fix: background services reports first repo
e2e: fix e2e failing tests

cmeessen commented on frontend/components/software/edit/repositories/SoftwareRepositoryModal.tsx:
```diff
@@ -0,0 +1,218 @@
+          >
+            <p>
+              The platform is, in most cases, based on the domain name in the provided url.
+              Our background services depend on platform definition for the communication with the platform api.
```
```suggestion
              Our background services depend on a platform definition for the communication with the platform API.
```

cmeessen reviewed (commented):
I strongly suggest to make the platform type always editable when adding a repository. This is because the detected platform depends on previous user input which might be wrong. This also implies that users must be able to edit an existing entry and change its platform type. Other than that there are only a couple of grammatical improvements from my side.

jmaassen commented:
For the information block at the side it currently says: 

<img width="497" height="228" alt="Screenshot_2025-11-11_09-34-21" src="https://github.com/user-attachments/assets/8502aba7-ac6c-48fd-b304-1eed404e2276" />

I would suggest changing this slightly to: 

Where can the source code be found?

Use the Add button to provide URL(s) to the repository of your software. The RSD background services will try to extract additional information about the software using the repository APIs. If needed, multiple repository URLs may be provided.

In may take up to 24 hour for the RSD background services to collecting additional information. See [documentation about scrapers](http://localhost/documentation/users/scrapers/) for more information.

dmijatovic committed:
chore: rename config object to cfg due to next app router name collision

dmijatovic committed:
fix: minimal update of repository scrapers (use id instead of software column in requests)
chore: bump scrapers version to v2.0.0 (breaking change in repository_url table)

dmijatovic commented:
I have updated textual remarks and fixed scraper requests. The platform drop-down is unlocked if the suggestion is based on previous entries, as suggested by Christian.

Further improvements of background service reports will be addressed in separate PR #1659

dmijatovic commented:
Closed in favor of #1666. All changes from this PR are included there.
